vlingo-eventjournal/src/main/java/io/vlingo/eventjournal/counter/events/CounterIncreasedEvent.java

```diff
@@ -0,0 +1,22 @@
+package io.vlingo.eventjournal.counter.events;
+
+import com.google.gson.Gson;
+import io.vlingo.symbio.Event;
+import io.vlingo.symbio.Metadata;
+import io.vlingo.symbio.State;
+
+import java.util.UUID;
+
+public class CounterIncreasedEvent {
+    private final UUID uuid;
+    public final int currentCounter;
+
+    public CounterIncreasedEvent(int currentCounter) {
+        this.uuid = UUID.randomUUID();
+        this.currentCounter = currentCounter;
+    }
+
+    public Event<String> toTextEvent() {
```

Comment:
This is convenient but we must be careful with examples not to set the wrong example :) I mean Hibernate and other tools have provided examples for 15-20 years that have sent developers down the road to dark brownness. That's why `vlingo-lattice` provides the `public interface SourceAdapter<S,R>` to separate all the serialization and migration/versioning issues. That said, this is the wrong place to introduce that because it would cause a circular dependency, but maybe provide some simple similar solution and point out that developers would normally use this `EventJournal` through `vlingo-lattice`.

Reply:
I will define an interface TextEventAdapter<S> (because this example only uses TextEvents :D) and change the code to remove the serialization logic from here.